File: Python/Arrays/10.TripletSum.py
from array import * 

def pairSum(arr, brr, crr):
    sum = 12

    for i in range (len(arr)):
        for j in range (len(brr)):
            for k in range (len(crr)):
                if arr[i] + brr[j] + crr[k]== sum:
                    print("Triplet is ", arr[i], " ", brr[j], " ", crr[k])
                    break

arr1 = array('i', [2, 3, 6, 8, 1, 9, 3])
arr2 = array('i', [4, 6, 9, 3, 2, 1, 7])
arr3 = array('i', [1, 3, 9, 7, 0, 1, 6])


pairSum(arr1, arr2, arr3)

File: CPP/Arrays/10.TripletSum.cpp
#include <bits/stdc++.h> 
using namespace std;


void tripletSum(vector<int> arr){
    int sum = 10;
    
    for(int i = 0; i< arr.size(); i++){
        for(int j =i+1; j< arr.size(); j++){
            for(int k = j+1; k< arr.size(); k++){
                if(arr[i] + arr[j] + arr[k] == sum){
                    cout<<"Triplt Is: "<<arr[i]<<" "<<arr[j]<<" "<<arr[k]<<endl;
                    break;
                }
            }
        }
    }
}


int main()
{
    vector<int> arr = {3, 5, 7, 9, 2, 4, 2, 6, 7, 2, 1, 4};
    
    tripletSum(arr);

    return 0;
}
